tool/tsh/db.go

```diff
@@ -109,6 +113,95 @@ func onListDatabases(cf *CLIConf) error {
 	return trace.Wrap(showDatabases(cf.SiteName, databases, activeDatabases, roleSet, cf.Format, cf.Verbose))
 }
 
+type databaseListing struct {
+	Proxy    string           `json:"proxy"`
+	Cluster  string           `json:"cluster"`
+	roleSet  services.RoleSet `json:"-"`
+	Database types.Database   `json:"database"`
+}
+
+func listDatabasesAllClusters(cf *CLIConf) error {
+	profile, profiles, err := client.Status(cf.HomePath, "")
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if profile != nil {
+		profiles = append(profiles, profile)
+	}
+
+	dbListings := make([]databaseListing, 0)
+	for _, p := range profiles {
```

Comment:
This logic repeats in each new method, what do you think about making a handy `forEachProfile` method would fetch all profiles, create a proper TeleportClient and call your method with it? Something like:

```go
func forEachProfile(cf *CLIConf, fn func (*client.TeleportClient) error) error {
  // fetch profiles, then:
  for _, profile := range profiles {
    // make teleport client for the profile
    // call "fn" passing it the client
  }
}

func listDatabasesAllClusters(cf *CLIConf) error {
  err := client.ForEachProfile(cf.HomePath, func(tc *client.TeleportClient) error {
    result, err := tc.ListDatabasesAllClusters(...)
    ...
  })
  ...
}
```